tools/generate-packed-assets

```diff
@@ -47,8 +47,16 @@ ASSETPACK_PLUGINS=$(grep 'AssetPack' lib/OpenQA/WebAPI.pm | awk -F '[()]' '{prin
 BUILD_CACHE="plugin AssetPack => { pipes => [qw(${ASSETPACK_PLUGINS})]} and app->asset->process()"
 BUILD_CACHE_OPTS='-Ilib/ -MMojolicious::Lite'
 
-# Allow command line options without quotes
-# shellcheck disable=SC2086
-env MOJO_MODE="test" perl $BUILD_CACHE_OPTS -e "$BUILD_CACHE" > /dev/null
-# shellcheck disable=SC2086
-env MOJO_MODE="development" perl $BUILD_CACHE_OPTS -e "$BUILD_CACHE" > /dev/null
+generate() {
+    # Allow command line options without quotes
+    # shellcheck disable=SC2086
+    env MOJO_MODE="$1" perl $BUILD_CACHE_OPTS -e "$BUILD_CACHE" > /dev/null
+}
+
+generate_retry() {
+    generate $1 || (sleep 1; generate $1 || exit 1) || (sleep 1; generate $1 || exit 1)
```

Comment:
In my last comment I said the other function needs quotes as well. Additionally, you don't need the explicit exit:
```suggestion
    generate "$1" || (sleep 1; generate "$1") || (sleep 1; generate "$1")
```

Spawning sub-shells also seems needless, e.g. one could easily write it as:
```suggestion
    [[ $2 ]] && sleep "$2"
    generate "$1" || generate "$1" "1" || generate "$1" "1"
```